console-framework-client/src/main/java/io/axoniq/console/framework/client/AxoniqConsoleRSocketClient.kt

```diff
@@ -46,46 +60,91 @@ class AxoniqConsoleRSocketClient(
 
-    private lateinit var rsocket: RSocket
-    private var connected = false
+    private var rsocket: RSocket? = null
+    private var lastSentHeartbeat = Instant.EPOCH
+    private var lastReceivedHeartbeat = Instant.now()
+    private var lastConnectionTry = Instant.EPOCH
+    private var connectionRetryCount = 0
+
+    init {
+        registrar.registerHandlerWithoutPayload(Routes.Management.HEARTBEAT) {
+            logger.debug("Received heartbeat from AxonIQ Console. Last one was: {}", lastReceivedHeartbeat)
+            lastReceivedHeartbeat = Instant.now()
+            lastReceivedHeartbeat
+        }
+    }
 
     override fun registerLifecycleHandlers(registry: Lifecycle.LifecycleRegistry) {
         registry.onStart(Phase.EXTERNAL_CONNECTIONS, this::start)
         registry.onShutdown(Phase.EXTERNAL_CONNECTIONS, this::dispose)
     }
 
+    /**
+     * Sends a message to the AxonIQ Console. If there is no connection active, does nothing silently.
+     * The connection will automatically be setup. Losing a few reports is no problem.
+     */
     fun send(route: String, payload: Any): Mono<Void> {
-        if (!connected) {
-            return Mono.empty()
-        }
         return rsocket
-            .requestResponse(encodingStrategy.encode(payload, createRoutingMetadata(route)))
-            .doOnError {
-                if (it.message?.contains("Access Denied") == true) {
-                    logger.info("Was unable to send call to AxonIQ Console since authentication was incorrect!")
-                }
-            }
-            .then()
+                ?.requestResponse(encodingStrategy.encode(payload, createRoutingMetadata(route)))
+                ?.then()
+                ?: Mono.empty()
     }
 
+    /**
+     * Starts the connection, and starts the maintenance task.
+     * The task will ensure that if heartbeats are missed the connection is killed, as well as re-setup in case
+     * the connection was lost. The task will do so with an exponential backoff with factor 2, up to a maximum of
+     * 60 seconds.
+     */
     fun start() {
-        this.scheduledReconnector = executor.scheduleWithFixedDelay({
-            if (!connected) {
-                logger.info("Reconnecting to AxonIQ Console...")
-                connect()
+        if(this.maintenanceTask != null) {
+            return
+        }
+        this.lastReceivedHeartbeat = Instant.now()
+        this.maintenanceTask = executor.scheduleWithFixedDelay(
+                this::ensureConnectedAndAlive,
+                initialDelay,
+                1000, TimeUnit.MILLISECONDS
+        )
+    }
+
+    private fun ensureConnectedAndAlive() {
+        if (isConnected()) {
+            if (!isAlive()) {
+                logger.info("Haven't received a heartbeat for {} seconds from AxonIQ Console. Reconnecting...", ChronoUnit.SECONDS.between(lastReceivedHeartbeat, Instant.now()))
+                rsocket?.dispose()
+                rsocket = null
```

Comment:
Why not invoke `disposeRsocket()` here?
```suggestion
                disposeRsocket()
```